local/compose/create.go

```diff
@@ -574,22 +574,25 @@ func (s *composeService) buildContainerVolumes(ctx context.Context, p types.Proj
 		return nil, nil, nil, err
 	}
 
-	// filter binds and volumes mount targets
 	volumeMounts := map[string]struct{}{}
 	binds := []string{}
+MOUNTS:
 	for _, m := range mountOptions {
-
-		if m.Type == mount.TypeVolume {
-			volumeMounts[m.Target] = struct{}{}
-			if m.Source != "" {
-				binds = append(binds, fmt.Sprintf("%s:%s:rw", m.Source, m.Target))
+		volumeMounts[m.Target] = struct{}{}
+		// `Bind` API is used when host path need to be created if missing, `Mount` is preferred otherwise
+		if m.Type == mount.TypeBind {
```

Comment:
To confirm: so we keep "bind" mounts only in the case of "createHostPath == true", otherwise default to the mount API even when `m.Type == mount.TypeBind`, is this correct ?

Reply:
yes indeed, which will only happen if user did define an explicit mount with full syntax, 
... which is probably less than 1% usages, everybody just uses `source:target` short syntax